packages/block-editor/src/components/contrast-checker/index.js

```diff
@@ -6,119 +6,124 @@ import namesPlugin from 'colord/plugins/names';
-	}, [ backgroundColor, textColor ] );
-
-	return (
-		<div className="block-editor-contrast-checker">
-			<Notice
-				spokenMessage={ null }
-				status="warning"
-				isDismissible={ false }
-			>
-				{ msg }
-			</Notice>
-		</div>
-	);
-}
-
 function ContrastChecker( {
 	backgroundColor,
 	fallbackBackgroundColor,
 	fallbackTextColor,
+	fallbackLinkColor,
 	fontSize, // font size value in pixels
 	isLargeText,
 	textColor,
+	linkColor,
 	enableAlphaChecker = false,
 } ) {
-	if (
-		! ( backgroundColor || fallbackBackgroundColor ) ||
-		! ( textColor || fallbackTextColor )
-	) {
+	const currentBackgroundColor = backgroundColor || fallbackBackgroundColor;
+
+	// Must have a background color.
+	if ( ! currentBackgroundColor ) {
 		return null;
 	}
-	const colordBackgroundColor = colord(
-		backgroundColor || fallbackBackgroundColor
-	);
-	const colordTextColor = colord( textColor || fallbackTextColor );
-	const textColorHasTransparency = colordTextColor.alpha() < 1;
+
+	const currentTextColor = textColor || fallbackTextColor;
+	const currentLinkColor = linkColor || fallbackLinkColor;
+
+	// Must have at least one text color.
+	if ( ! currentTextColor && ! currentLinkColor ) {
+		return null;
+	}
+
+	const colordBackgroundColor = colord( currentBackgroundColor );
 	const backgroundColorHasTransparency = colordBackgroundColor.alpha() < 1;
-	const hasTransparency =
-		textColorHasTransparency || backgroundColorHasTransparency;
-	const isReadable = colordTextColor.isReadable( colordBackgroundColor, {
+
+	// If there's only one color passed, store in `singleTextColor`.
+	const singleTextColor =
+		currentTextColor && currentLinkColor
+			? null
+			: currentTextColor || currentLinkColor;
+
+	const colordTextColor = singleTextColor
+		? colord( singleTextColor )
+		: colord( currentTextColor );
+	const colordLinkColor = colord( currentLinkColor );
+
+	// Transparency.
+	const textColorHasTransparency =
+		currentTextColor && colordTextColor.alpha() < 1;
+	const linkColorHasTransparency =
+		currentLinkColor && colordLinkColor.alpha() < 1;
+
+	// Text size.
+	const isReadableOptions = {
 		level: 'AA',
 		size:
 			isLargeText || ( isLargeText !== false && fontSize >= 24 )
 				? 'large'
 				: 'small',
-	} );
+	};
+
+	// Readability.
+	const isTextColorReadable =
+		currentTextColor &&
+		colordTextColor.isReadable( colordBackgroundColor, isReadableOptions );
+
+	const isLinkColorReadable =
+		currentLinkColor &&
+		colordLinkColor.isReadable( colordBackgroundColor, isReadableOptions );
+
+	// Flag to warn about transparency only if the text is otherwise readable according to colord
+	// to ensure the readability warnings take precedence.
+	let shouldShowTransparencyWarning = false;
 
 	// Don't show the message if the text is readable AND there's no transparency.
 	// This is the default.
-	if ( isReadable && ! hasTransparency ) {
-		return null;
-	}
-
-	if ( hasTransparency ) {
+	if ( ! textColorHasTransparency && ! linkColorHasTransparency ) {
```

Comment:
Much of this could probably be a hook of some description, or at least extracted for now.

Reply:
This is testing well for me. I agree that this might benefit from pulling out into some methods to remove some of the `if` nesting, but I struggled to see a quick/easy way of doing this without making it look just as complicated.

Reply:
Thanks for testing. Yeah, it's fairly convoluted. What's there wasn't my first attempt either 😆 

The alpha rules really throw a pineapple on the pizza in that they're not part of the official "readable" test. Colord treats rgba like rgb.